### ISSUE TITLE
Support TLS between Operator and DB cluster

### ISSUE THREAD

**Issue by muhaook** [CrunchyData/postgres-operator#901]
4 features need to be modified in order to use TLS between postgres operator and DB:

**1/ maintain PostgreSQL users and passwords**

I used operator to change user's password, it succeeded. Looking into crunchy source code, I found operator is using pq (https://github.com/lib/pq) as the db driver to connect to PostgreSQL db. The connection is like:
```
conn, err = sql.Open("postgres", "sslmode=disable user="+info.Username+" host="+info.Hostip+" port="+info.Port+" dbname="+info.Database+" password="+info.Password)
```
The sslmode=disable is hardcoded in operator. This needs to be changed if we want to use TLS to connect to DB.

**2/ apply SQL policies to PostgreSQL clusters**

I succeeded using operator to apply a SQL policy to DB. But in operator, ExecPolicy uses runpsql.sh to create policy in Postgres DB. In runpsql.sh, a psql cmdline is executed with password.
```
PGPASSWORD=$1 psql -p 5432 -h $2 -U postgres postgres
```
This needs to be changed if we want to use SSL/TLS.

**3/ load csv/json file into DB**
looks once postgres operator receives a data loading request from pgo client, it starts seperate docker container to perform the actual data loading. The docker container's name is pgo-load. In this container, username/password is used to connect to DB and then run 'psql copy' to complete the data loading.
```
cd /tmp
cat >> ".pgpass" <<-EOF
*:*:*:*:${DB_PASS}
EOF
chmod 0600 .pgpass
export PGPASSFILE=/tmp/.pgpass
echo "\COPY $TABLE_TO_LOAD  FROM '/pgdata/$FILE_PATH' WITH (FORMAT csv);" > $OUTFILE
psql -U $DB_USER -h $DB_HOST $DB_DATABASE -f $OUTFILE
echo "pgo-load has ended!"
```
This needs to be changed if we want to use SSL/TLS.

**4/ dump**
'Dump' is a logical backup on postgres DB. Once postgres operator receives a dump request, it starts a seperate docker container to perform the dump. I found the source code of the 'pgdump' container at github.com/crunchy-containers. In the code, username/password is used. We can not change configration to enable TLS.
```
cat >> "${PGPASSFILE?}" <<-EOF
*:*:*:${PGDUMP_USER?}:${PGDUMP_PASS?}
EOF
${PGROOT?}/bin/pg_dump ${conn_opts?} ${PGDUMP_CUSTOM_OPTS?} ${output_opts?}
```
This needs to be changed if we want to use SSL/TLS.

**Comment by jkatz** [comment]
See #1180 for current discussion around TLS support. A few of these issues were fixed in the `v4.2.0` release, and we will continue to ensure we can fully support TLS with PostgreSQL.

**Comment by jkatz** [comment]
See #1288 

Closing this in favor of #1180
